### ISSUE TITLE
Why Use Old Masks for Evaluation During Global Model Testing in Each Round?

### ISSUE THREAD

**Issue by Dxpsk** [git-disl/Lockdown#2]
### Question:
In the following code snippet:

```python
if args.method == "lockdown" or args.method == "fedimp":
    old_mask = [copy.deepcopy(agent.mask) for agent in agents]
--------------------------------------------------------------------------------------------------
if args.method == "lockdown" :
    test_model = copy.deepcopy(global_model)
    for name, param in test_model.named_parameters():
        mask = 0
        for id, agent in enumerate(agents):
            mask += old_mask[id][name].to(args.device)   

        global_mask[name] = torch.where(mask.to(args.device) >= args.theta, torch.ones_like(mask), torch.zeros_like(mask))

        param.data = torch.where(mask.to(args.device) >= args.theta, param,
                                 torch.zeros_like(param))
        
        logging.info(torch.sum(mask.to(args.device) >= args.theta) / torch.numel(mask))
    val_loss, (val_acc, val_per_class_acc), _ = utils.get_loss_n_accuracy(test_model, criterion,
                                                                          val_loader,
                                                                          args, rnd, num_target)
```
Why is the old mask of clients (from before the current round's updates) being used during global model evaluation in each round? Wouldn't it be more appropriate to use the updated masks? What is the rationale behind this design choice?

Additionally, I noticed that the code does not update the global mask, which makes it incompatible with the omniscient attack. To address this, I suggest adding the following line to update the global mask:

```python
global_mask[name] = torch.where(mask.to(args.device) >= args.theta, torch.ones_like(mask), torch.zeros_like(mask))
```

Thank you for any insights!

**Comment by huangtiansheng** [comment]
Hi,

Thanks for reaching out. Honestly I couldn't remember the exact reason why I choose to use the old mask instead of the updated mask in the current round. I think both of the solutions works and only incur minor difference. For the two implementation, the results should be identical, except that the order of the appearance of the results are one round different.

**Comment by Dxpsk** [comment]
Thank you for your quick response and clarification.
